core/src/main/java/io/questdb/cairo/sql/async/WorkStealingStrategy.java

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.sql.async;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Defines work stealing behavior of the owner thread for queries processed in parallel,
+ * such as queries with parallel filters (WHERE) and GROUP BY queries.
+ */
+public interface WorkStealingStrategy {
+
+    WorkStealingStrategy of(AtomicInteger startedCounter);
+
+    /**
+     * Returns true if the query owner thread should spin instead of
+     * trying to pick up one of the published tasks.
+     */
+    boolean shouldStealWork(int finishedCount);
```

Comment:
Naming is not obvious, maybe read something like "should the stealing work" instead of "should the work be stolen"?

Reply:
How about `shouldSteal`?

Reply:
Renamed in af191f0866516b627b9b38bcc7b85d2030487a5b